### DISCUSSION TITLE
Search results when clicked does not points to the page in Cloudflare Pages

### DISCUSSION THREAD

**Discussion by mahadevan** [CaiJimmy/hugo-theme-stack#940] [Q&A]
Hi,

I deployed my site using the hugo-theme-stack template. I copied .md files from my earlier sites from posts into it. 

Now, every post link works fine when I click it through links. But when I use the search module, it shows results correctly, but **when I click it, it gives me Cloudflare error** `Nothing is here yet. If the project exists, it may not be ready yet. Please check back later.`. I think it is equivalent to 404. 

But the same link works if I go through other links like pages, tag, categories etc. But when I use search it gives this error. 

![image](https://github.com/CaiJimmy/hugo-theme-stack/assets/2020804/e574be18-c5c6-451f-bd51-43f436da1cf2)

*nb: Cloudflare builds on /public directory. when I check public/search/index.json it shows all files in it correctly.* 

Also when I paste the same link in the address bar it gives the same Cloudflare error. The link works if comes through other links though.

What should I do to fix this so that i can paste the link / search the page and the page shows up?

Please help.

**UPDATE : It is working perfectly locally. But search still not working on remote.**

**Comment by rmdhnreza** [comment]
Do you use the latest version of Hugo at Cloudflare?
Does the Cloudflare build log contain any errors?

**Reply by mahadevan** [reply]
I use the latest version of Hugo. I set the new Variable `HUGO_VERSION` as the latest version.
No error in the Cloudflare log.
Links work only if they come through other links. If it comes through search, it does list it but does not redirect to that link when clicked. Only search has problem.

**Reply by mahadevan** [reply]
It is working perfectly locally. But search still not working on remote.

**Comment by CaiJimmy** [comment]
Sorry for the late response. Is it fixed now? If not, please include the URL to the page with the problem, otherwise it will be impossible for us to find the cause.

**Reply by mahadevan** [reply]
Local Search is not working on cloudflare pages. But it is working on Github Pages. So I shifted to Github Pages.

**Reply by CaiJimmy** [reply]
That's strange. I have deployed `hugo-theme-stack-starter` on CloudFlare pages, and the search function appears to work correctly: https://stack-theme.pages.dev/search/. My best guess is that the `baseURL` configuration for the site was not set correctly, as the search results use absolute links.
